sparse_strips/vello_toy/src/svg.rs

```diff
@@ -0,0 +1,254 @@
+// Copyright 2025 the Vello Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! A very basic SVG renderer built on top of `usvg` and `vello_cpu`.
+
+#![allow(
+    clippy::cast_possible_truncation,
+    reason = "this is only a debug tool, so we can ignore them"
+)]
+
+use clap::Parser;
+use image::codecs::png::PngEncoder;
+use image::{ExtendedColorType, ImageEncoder};
+use peniko::Fill;
+use peniko::color::AlphaColor;
+use peniko::kurbo::Stroke;
+use std::io::Cursor;
+use std::path;
+use std::path::Path;
+use std::time::{Duration, Instant};
+use usvg::tiny_skia_path::PathSegment;
+use usvg::{Node, Paint, PaintOrder};
+use vello_common::kurbo::{Affine, BezPath};
+use vello_cpu::{Pixmap, RenderContext};
+
+fn main() {
+    let args = Args::parse();
+    let scale = args.scale;
+    let path = Path::new(&args.path);
+    let file_name = path.file_name().unwrap().to_str().unwrap();
+
+    let svg = std::fs::read_to_string(path).expect("error reading SVG file");
+    let tree = usvg::Tree::from_str(&svg, &usvg::Options::default()).unwrap();
+    let width = (tree.size().width() * scale).ceil() as u16;
+    let height = (tree.size().height() * scale).ceil() as u16;
+
+    let mut num_iters = 0;
+    let mut pixmap;
+    let mut ctx;
+    let mut runtime = Duration::default();
+
+    loop {
+        let mut sctx = SVGContext::new_with_scale(scale as f64);
+        pixmap = Pixmap::new(width, height);
+        ctx = RenderContext::new(width, height);
+
+        let start = Instant::now();
+
+        render_tree(&mut ctx, &mut sctx, &tree);
+        ctx.render_to_pixmap(&mut pixmap);
+
+        runtime += start.elapsed();
+        num_iters += 1;
+
+        if runtime.as_millis() > args.runtime as u128 {
+            break;
+        }
+    }
+
+    let avg_runtime = (runtime.as_millis() as f32) / (num_iters as f32);
+
+    eprintln!(
+        "ran {} iterations, with an average runtime of {}ms to render {}.",
+        num_iters, avg_runtime, file_name
+    );
+
+    write_pixmap(&mut pixmap);
+}
+
+fn write_pixmap(pixmap: &mut Pixmap) {
+    pixmap.unpremultiply();
+
+    let mut png_data = Vec::new();
+    let cursor = Cursor::new(&mut png_data);
+    let encoder = PngEncoder::new(cursor);
+    encoder
+        .write_image(
+            pixmap.data(),
+            pixmap.width() as u32,
+            pixmap.height() as u32,
+            ExtendedColorType::Rgba8,
+        )
+        .expect("failed to encode image");
+
+    let path = path::absolute("svg.png").unwrap();
+    std::fs::write(&path, png_data).unwrap();
+    eprintln!("saved rendered SVG to '{}'", path.display());
+}
+
+#[derive(Parser, Debug)]
+struct Args {
+    /// How much to scale the output image.
+    #[arg(long, default_value_t = 1.0)]
+    pub scale: f32,
+    /// The path to the SVG file.
+    #[arg(long)]
+    pub path: String,
+    /// The target runtime for calculating how many iterations to run, in milliseconds.
+    #[arg(long, default_value_t = 2000)]
```

Comment:
I'd prefer for this to default to running a single iteration, i.e.:
```suggestion
    /// For simple benchmarking, our current recommended value is 2000 (i.e. 2 seconds).
    #[arg(long, default_value_t = 0)]
```

I do agree that the benchmarking is a useful tool, but I don't think it should be the default.
(I can also see arguments for a `benchmark` flag, but that's probably too much complication for now)